src/agent/main.cpp

```diff
@@ -232,8 +232,8 @@ static int realmain(int argc, char *argv[])
             break;
 
         case OTBR_OPT_DEBUG_LEVEL:
-            logLevel = atoi(optarg);
-            VerifyOrExit(logLevel >= OTBR_LOG_EMERG && logLevel <= OTBR_LOG_DEBUG, ret = EXIT_FAILURE);
+            logLevel = static_cast<otbrLogLevel>(atoi(optarg));
+            VerifyOrExit(logLevel >= OTBR_LOG_LEVEL_CRIT && logLevel <= OTBR_LOG_LEVEL_DEBG, ret = EXIT_FAILURE);
```

Comment:
Should we treat `logLevel` > `OTBR_LOG_LEVEL_DEBG` same as `OTBR_LOG_LEVEL_DEBG` when parsing the arguments? Otherwise there will be many unexpected failures after this PR. 

```suggestion
            VerifyOrExit(logLevel >= OTBR_LOG_LEVEL_CRIT, ret = EXIT_FAILURE);
            logLevel = std::min(logLevel, OTBR_LOG_LEVEL_DEBG);
```